doc/source/status.rst

```diff
@@ -1,36 +1,243 @@
+
+SubscriptionStatus
+------------------
+
+The :class:`~ophyd.status.SubscriptionStatus` is a special Status object that
+correctly and succinctly handles a common use case, wherein the Status object
+is marked finished based on some ophyd event. It reduces this:
+
+.. code:: python
+
+   from ophyd import Device, Component, DeviceStatus
+
+   class MyToyDetector(Device):
+       ...
+       # When set to 1, acquires, and then goes back to 0.
+       acquire = Component(...)
+
+       def trigger(self):
+           def check_value(old_value, value, **kwargs):
+               if old_value == 1 and value == 0:
+                   status.set_finished()
+                   # Clear the subscription.
+                   sself.acquire.clear_sub(check_value)
+
+           status = DeviceStatus(self.acquire)
+           self.acquire.subscribe(check_value)
+           self.acquire.set(1)
+           return status
+
+to this:
+
+.. code:: python
+
+   from ophyd import Device, Component, SubscriptionStatus
+
+   class MyToyDetector(Device):
+       ...
+       # When set to 1, acquires, and then goes back to 0.
+       acquire = Component(...)
+
+       def trigger(self):
+           def check_value(old_value, value, **kwargs):
+               if old_value == 1 and value == 0:
+                   status.set_finished()
+
+           status = SubscriptionStatus(self.acquire, check_value)
+           self.acquire.set(1)
+           return status
+
+Note that ``subscribe`` and ``clear_sub`` are gone; they are handled
+automatically, internally. See :class:`~ophyd.status.SubscriptionStatus` for
+additional options.
+
+Partial Progress Updates
+------------------------
+
+Some Status objects provide an additional method named ``watch``, as in
+:meth:`~ophyd.status.MoveStatus.watch`, which can be used to subscribe to
+*incremental* progress updates suitable for building progress bars. See
+:doc:`bluesky:progress-bar` for one application of this feature.
+
+The ``watch`` method accepts a callback which must accept the following
+parameters as optional keyword arguments:
+
+* ``name``
+* ``current``
+* ``initial``
+* ``target``
+* ``unit``
+* ``precision``
+* ``fraction``
+* ``time_elapsed``
+* ``time_remaining``
+
+The callback may receive a subset of these depending on how much we can know
+about the progress of a particular action. In the case of
+:obj:`ophyd.status.MoveStatus` and
+:obj:`ophyd.areadetector.trigger_mixins.ADTriggerStatus`, we know a lot, from
+which one can build a frequently-updating progress bar with a realistic
+estimated time of completion. In the case of a generic
+:obj:`ophyd.status.DeviceStatus`, we only know the name of the assocated
+Device, when the action starts, and when the action ends.
+
+Status API details
+------------------
+
+.. autoclass:: ophyd.status.StatusBase
+   :members:
+
+In addition we provide specialized subclasses that know more about the object
 they are tied to.
 
-.. autosummary::
```

Comment:
I like a page-per-class a bit better than having them all inline, but :man_shrugging:

Reply:
The way that h11 and trio flow has converted me to having them inline is helpful when the number of classes is small. I would definitely not change the bluesky plans page to take this approach.